Aztec/Classes/TextKit/TextStorage.swift

```diff
@@ -356,6 +360,13 @@ open class TextStorage: NSTextStorage {
 
             processUnderlineDifferences(in: domRange, betweenOriginal: sourceStyle, andNew: targetStyle)
         case NSAttachmentAttributeName:
+            if sourceValue is LineAttachment || targetValue is LineAttachment {
```

Comment:
We should avoid checking this here with an `if` block.

It's a precondition of this method (`processAttributesDifference`) that either the source value or the target value must be non-nil (in the same way it's a precondition for the other attributes).

The rationale is that `enumerateAttributeDifferences()` is responsible of only signaling real differences: an attribute that doesn't change is NOT a difference, and would signal that the method is not working right.

If you want to make sure that this precondition is true, an assertion would be more appropriate.  For instance we could check at the start of this method:

```swift
assert(sourceValue != nil || targetValue != nil)
```

Or you could add a more specific assertion for your attribute.

```swift
assert(sourceValue is LineAttachment || targetValue is LineAttachment)
```

* My mistake for not documenting in this method that it is actually a precondition.